device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -226,9 +230,38 @@ public void onMessageSent(Message message, String deviceId, Throwable e)
                     this.handleMessageException(packet, new TransportException(e));
                 }
             }
+
+            try {
+                String messageId = message.getCorrelationId();
+                if (messageId != null && correlationCallbacks.containsKey(messageId))
+                {
+                    Object context = null;
+                    if (correlationContext.containsKey(messageId))
+                    {
+                        context = correlationContext.get(messageId);
```

Comment:
map.get(object) returns null if the key isn't present, so you can simplify this to

```java
Object context = correlationContext.get(messageId);
```